py/server/deephaven/json/__init__.py

```diff
@@ -0,0 +1,1221 @@
+#
+# Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+#
+
+"""The deephaven JSON module presents a declarative and composable configuration layer for describing the structure of a
+JSON value. It is meant to have sane defaults while also providing finer-grained configuration options for typical
+scenarios. The primary purpose of this module is to provide a common layer that various consumers can use to parse JSON
+values into appropriate Deephaven structures. As such (and by the very nature of JSON), these types represent a superset
+of JSON. This module can also service other use cases where the JSON structuring is necessary (for example, producing a
+JSON value from a Deephaven structure).
+"""
+
+import jpy
+from dataclasses import dataclass, field
+from datetime import datetime
+from enum import Enum
+from typing import Dict, List, Union, Tuple, Optional, Literal, Any
+
+from deephaven import dtypes
+from deephaven._wrapper import JObjectWrapper
+from deephaven.time import to_j_instant
+from deephaven._jpy import strict_cast
+
+
+__all__ = [
+    "string_",
+    "bool_",
+    "char_",
+    "byte_",
+    "short_",
+    "int_",
```

Comment:
Do we want to be more explicit/pythonic in type naming? `int` is clear and well defined in Java, but less so in Python. `int32`?

Reply:
We've decided to append `_val`. This is similar to how `int_col` works for example.